heart_animation_2/lib/animation.dart

```diff
@@ -0,0 +1,109 @@
+import 'package:flutter/material.dart';
+import 'package:heart_animation_2/constants.dart';
+
+class HeartAnimation extends StatefulWidget {
+  const HeartAnimation({super.key});
+
+  @override
+  State<HeartAnimation> createState() => _HeartAnimationState();
+}
+
+class _HeartAnimationState extends State<HeartAnimation> with TickerProviderStateMixin {
+  // [_controller] is for inner animation effect.
+  late AnimationController _controller;
+
+  // [_outerBoxController] is outer circle animation controller.
+  late AnimationController _outerBoxController;
+
+  // inner circle animation
+  late Animation<double> _animation;
+
+  @override
+  void initState() {
+    _controller = AnimationController(
+      vsync: this,
+      duration: const Duration(milliseconds: 600),
+      reverseDuration: const Duration(milliseconds: 400),
+    );
+
+    _outerBoxController = AnimationController(
+      vsync: this,
+      duration: const Duration(milliseconds: 300),
+      lowerBound: 0.0,
+      upperBound: 0.1,
+    );
+
+    _animation = CurvedAnimation(
+      parent: _controller,
+      curve: const ElasticOutCurve(0.8),
+    );
+
+    super.initState();
```

Comment:
Let's have this super.initState() line as the first line.
[Read here why](https://stackoverflow.com/questions/58400001/should-i-call-super-initstate-at-the-end-or-at-the-beginning/58400324#58400324)

Reply:
Yeah sure.
Thanks for the StackOverflow link.